Branch-SDK/src/main/java/io/branch/referral/util/BranchEvent.java

```diff
@@ -256,6 +269,175 @@ public boolean logEvent(Context context) {
+                    .setProductId(productId)
+                    .setProductType(BillingClient.ProductType.INAPP)
+                    .build();
+
+            QueryProductDetailsParams.Product subsProduct = QueryProductDetailsParams.Product.newBuilder()
+                    .setProductId(productId)
+                    .setProductType(BillingClient.ProductType.SUBS)
+                    .build();
+
+            productList.add(inAppProduct);
+            subsList.add(subsProduct);
+        }
+
+        QueryProductDetailsParams queryProductDetailsParams =
+                QueryProductDetailsParams.newBuilder()
+                        .setProductList(productList)
+                        .build();
+
+        QueryProductDetailsParams querySubsProductDetailsParams =
+                QueryProductDetailsParams.newBuilder()
+                        .setProductList(subsList)
+                        .build();
+
+        PurchasesUpdatedListener purchasesUpdatedListener = new PurchasesUpdatedListener() {
+            @Override
+            public void onPurchasesUpdated(BillingResult billingResults, List<Purchase> purchases) {
+
+            }
+        };
+
+        BillingClient billingClient = BillingClient.newBuilder(context)
+                .setListener(purchasesUpdatedListener)
+                .enablePendingPurchases()
+                .build();
+
+        billingClient.startConnection(new BillingClientStateListener() {
+            @Override
+            public void onBillingSetupFinished(BillingResult billingResult) {
+                if (billingResult.getResponseCode() == BillingClient.BillingResponseCode.OK) {
+
+
+                    billingClient.queryProductDetailsAsync(querySubsProductDetailsParams, (billingQueryResult, productDetailsList) -> {
+                        if (billingQueryResult.getResponseCode() == BillingClient.BillingResponseCode.OK) {
+                            List<BranchUniversalObject> buos = new ArrayList<>();
+                            CurrencyType currency = null;
+                            double revenue = 0.00;
+                            String purchasedProductType = "";
+
+                            for (ProductDetails product : productDetailsList) {
+                                double price;
+                                int quantity;
+
+                                purchasedProductType = "Subscription";
+                                currency = CurrencyType.valueOf(product.getSubscriptionOfferDetails().get(0).getPricingPhases().getPricingPhaseList().get(0).getPriceCurrencyCode());
+                                price = product.getSubscriptionOfferDetails().get(0).getPricingPhases().getPricingPhaseList().get(0).getPriceAmountMicros() / 1_000_000;
+                                quantity = 1;
+                                revenue += price * quantity;
+
+
+                                BranchUniversalObject buo = new BranchUniversalObject()
```

Comment:
Separate out this conversion logic into a function or utility class for easier testing.